### DISCUSSION TITLE
gcTime for queryClient.setQueryData

### DISCUSSION THREAD

**Discussion by james-wee-iress** [TanStack/query#6488] [Q&A]
<img width="840" alt="image" src="https://github.com/TanStack/query/assets/97071346/01175b5e-990a-447d-9e69-8bf7cccdb56b">

Based on this part of the documentation, it seems that `setQueryData` will have a 5 minute `gcTime` if not picked up by a useQuery hook. But we have a use case where the server data will not go stale and we do not know when the server data is going to be used, but the possibility of it being used is very high. 

So, I wonder if there is a way to `setQueryData` with a custom `gcTime` like:
```
setQueryData(queryKey, data, {
  gcTime: Infinity
})
```

**Comment by TkDodo** [comment]
it will pick up the default gcTime, wich is 5 minutes, unless you specify a different default globally or per query key with `queryClient.setQueryDefaults`

**Reply by nklhtv** [reply]
@TkDodo Is there a way to specify the default gcTime for exact queries only?

The defaults set with `queryClient.setQueryDefaults(['items'], { ... })` will affect queries like `['items', 5]`

If we could pass something like `{exact: true}` like we can in queryClient.invalidateQueries

**Reply by TkDodo** [reply]
no, it's always a fuzzy match.
